### PR TITLE
Update `blacklist` and `whitelist` keywords

### PR DESCRIPTION
**Pull request by trungleduc**
## References
Closes https://github.com/voila-dashboards/voila/issues/1366

## User-facing changes

- Keywords with `blacklist` are replaced by `denylist`
- Keywords with `whitelist` are replaced by `allowlist`
- A `DeprecationWarning` message is raised on using the old keywords.

## Backwards-incompatible changes

N/A

### CODE REVIEW

File: voila/__init__.py [hunk 1]
```diff
@@ -10,6 +10,9 @@
 from ._version import __version__  # noqa
 from .server_extension import _load_jupyter_server_extension  # noqa
 from .server_extension import load_jupyter_server_extension  # noqa
+import warnings
+
+warnings.filterwarnings("default", category=DeprecationWarning, module="traitlets")
```
Comment (martinRenou) [inline voila/__init__.py:15]: Mmmh, Maybe we should not do that and fix the deprecation warnings instead?

File: voila/configuration.py [hunk 2]
```diff
@@ -52,30 +54,62 @@ class VoilaConfiguration(traitlets.config.Configurable):
...
+    )
+
+    @validate("file_whitelist")
+    def _valid_file_whitelist(self, proposal):
+        warn(
+            "Deprecated, use VoilaConfiguration.file_allowlist instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        return proposal["value"]
```
Comment (martinRenou) [inline voila/configuration.py:82]: May I suggest another approach for this?

```python
@property
def file_whitelist(self):
   warn(
        "Deprecated, use VoilaConfiguration.file_allowlist instead.",
        DeprecationWarning,
        stacklevel=2,
    )
   return self.file_allowlist 


@file_whitelist.setter
def file_whitelist(self, value):
   warn(
        "Deprecated, use VoilaConfiguration.file_allowlist instead.",
        DeprecationWarning,
        stacklevel=2,
    )
   self.file_allowlist = value
```

That way we can get rid of the

```python
warnings.filterwarnings("default", category=DeprecationWarning, module="traitlets")
```

?

Comment (martinRenou) [reply voila/configuration.py:82]: My suggestion applies if we consider going from a traitlet to a property is not backward incompatible

File: voila/__init__.py (hunk 1 above)
Comment (trungleduc) [reply voila/__init__.py:15]: Nope, the `DeprecationWarning` is hidden by default https://docs.python.org/3/library/exceptions.html#DeprecationWarning

File: voila/configuration.py (hunk 2 above)
Comment (trungleduc) [reply voila/configuration.py:82]: By doing so, users can not set the allow/deny list from the CLI or the `voila.json` file

**Review by martinRenou — APPROVED** [review]
Thanks!

**Pull request merged**